pkg/route/api/allocator/allocator.go

```diff
@@ -0,0 +1,52 @@
+package allocator
+
+import (
+	"fmt"
+	"regexp"
+
+	"code.google.com/p/go-uuid/uuid"
+	routeapi "github.com/openshift/origin/pkg/route/api"
+)
+
+// This should be something we get from config but we would still need a
+// default name if nothing's passed. The first pass uses the default name.
+// Would be better if we could use "v3.openshift.app", someone bought that!
+const DEFAULT_DNS_SUFFIX = "v3.openshift.com"
+
+// Invalid host name characters or not the <let-or-digit-or-hyphen> and
+// '.' separator as defined in rfc1123 (via rfc952)
+var name_RE, _ = regexp.Compile("[^A-Za-z0-9-.]+")
+
+func allocateShard(route *routeapi.Route) string {
+	//  For later, allocate names pointing to different shards.
+	shard := DEFAULT_DNS_SUFFIX
```

Comment:
Allocation (once we have shards) will be a controller and a separate process.

Turn this into a type, take this as a value passed in, and then do it that way.